common/v2/types/global.d.ts

```diff
@@ -2,3 +2,40 @@
 // enable a fast change once the typings are fixed
 // https://spectrum.chat/styled-components/general/typescript-refs~5857d917-966e-4a71-940f-524206896f43
 type SCref = any;
+
+export enum SecureWalletName {
+  WEB3 = 'web3',
+  LEDGER_NANO_S = 'ledgerNanoS',
+  TREZOR = 'trezor',
+  SAFE_T = 'safeTmini',
+  PARITY_SIGNER = 'paritySigner'
+}
+
+export enum HardwareWalletName {
+  LEDGER_NANO_S = 'ledgerNanoS',
+  TREZOR = 'trezor',
+  SAFE_T = 'safeTmini'
+}
+
+export enum InsecureWalletName {
+  PRIVATE_KEY = 'privateKey',
+  KEYSTORE_FILE = 'keystoreFile',
+  MNEMONIC_PHRASE = 'mnemonicPhrase'
+}
+
+export enum MiscWalletName {
+  VIEW_ONLY = 'viewOnly'
+}
+
+export enum DefaultWalletName {
+  DEFAULT = 'walletsList'
+}
+
+export const walletNames = getValues(
+  SecureWalletName,
+  HardwareWalletName,
+  InsecureWalletName,
+  MiscWalletName
+);
+
+export type WalletName = SecureWalletName | InsecureWalletName | MiscWalletName;
```

Comment:
Don't all the wallet names already exist here: https://github.com/MyCryptoHQ/MyCrypto/pull/2599/files#diff-c319aeb165f2fd1efff5f018a6511d56R105 ? Why do we need another duplication of these?

Reply:
I was trying to find this 😅